### PR TITLE
Add support for 'duration' attributes.

### PR DESCRIPTION
**Pull request by geeknoid**
Also includes minimal support for the updated
QuotaDescriptor, pending deep integration.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/248)

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -155,7 +155,7 @@ new_git_or_local_repository(
     name = "com_github_istio_api",
     build_file = "BUILD.api",
     path = "../api",
-    commit = "7917b2d041a9ef931e242ae58caa584158eb4cf3", # Jan 31, 2017 (no releases)
+    commit = "2df567af4b8cf0727c59f48cf81d3268fe7ad628", # Jan 31, 2017 (no releases)
```
Comment (douglas-reid) [inline WORKSPACE]: is the date correct now?

File: cmd/client/tmp [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (douglas-reid) [inline cmd/client/tmp]: this seems like something that should not be included.

File: pkg/api/grpcServer_test.go [hunk 3]
```diff
@@ -115,7 +115,7 @@ func (ts *testState) Report(ctx context.Context, tracker attribute.Tracker, requ
 
 func (ts *testState) Quota(ctx context.Context, tracker attribute.Tracker, request *mixerpb.QuotaRequest, response *mixerpb.QuotaResponse) {
 	response.RequestIndex = request.RequestIndex
-	response.Result = newStatus(code.Code_UNIMPLEMENTED)
+	//	response.Result = newStatus(code.Code_UNIMPLEMENTED)
```
Comment (douglas-reid) [inline pkg/api/grpcServer_test.go]: is this meant to be included?

File: pkg/api/handler.go [hunk 4]
```diff
@@ -138,8 +138,8 @@ func (h *handlerState) Report(ctx context.Context, tracker attribute.Tracker, re
 // Quota performs 'quota' function corresponding to the mixer api.
 func (h *handlerState) Quota(ctx context.Context, tracker attribute.Tracker, request *mixerpb.QuotaRequest, response *mixerpb.QuotaResponse) {
 	response.RequestIndex = request.RequestIndex
-	status := h.execute(ctx, tracker, request.AttributeUpdate, config.QuotaMethod)
-	response.Result = newStatus(code.Code(status.Code))
+	//	status := h.execute(ctx, tracker, request.AttributeUpdate, config.QuotaMethod)
```
Comment (douglas-reid) [inline pkg/api/handler.go]: meant to be included?

File: WORKSPACE (hunk 1 above)
Comment (geeknoid) [inline WORKSPACE]: Updated.

File: cmd/client/tmp (hunk 2 above)
Comment (geeknoid) [inline cmd/client/tmp]: Hmm, I don't know where that came from.

Gone.

File: pkg/api/handler.go (hunk 4 above)
Comment (geeknoid) [inline pkg/api/handler.go]: I meant it, but it wasn't a great idea :-) Improved slightly. This is all pending the finalization of the API design and the real implementation of course.

**Comment by geeknoid** [comment]
PTAL.

**Pull request merged**